### PR TITLE
Implemented separation of concerns into layers 🎉

### PR DESCRIPTION
**Pull request by portellaa**

### CODE REVIEW

File: Sources/App/Store/CRUDStore.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    func create(model: M) -> SignalProducer<M, Swift.Error>
+    func read(by id: M.ID) -> SignalProducer<M?, Swift.Error>
+    // TODO: Support perhaps support keypath queries
+//    func retrieve(by key: KeyPath<M, Encodable>) -> SignalProducer<M?, E>
+    func delete(by id: M.ID) -> SignalProducer<M, Swift.Error>
+}
+
+extension CRUDStore {
+    func create(model: M) -> SignalProducer<M, Swift.Error> {
+        return model.save(on: databaseConnectable).reactive
```
Comment (tallnato) [inline Sources/App/Store/CRUDStore.swift:19]: So that it stays the same as `read`, maybe `.reactive` could be in a new line? 💄

**Review by tallnato — APPROVED** [review]
LGTM 🚀

File: Sources/App/configure.swift [hunk 2]
```diff
@@ -18,7 +16,7 @@ public func configure(_ config: inout Config, _ env: inout Environment, _ servic
     // Configure MySQL
     // TODO: change later to support environments
     let mysqlDatabaseConfig = MySQLDatabaseConfig(hostname: "127.0.0.1",
-                                                  port: 32773,
+                                                  port: 32768,
                                                   username: "root",
                                                   password: "youclap-sql-pw",
                                                   database: "youclap-dev",
```
Comment (ruimartinsptl) [inline Sources/App/configure.swift:22]: All these parameters should be variables, introduced as environment variables, right?

**Review by ruimartinsptl — APPROVED** [review]
LGFM, caution with database credentials in code...

File: Sources/App/configure.swift (hunk 2 above)
Comment (ruimartinsptl) [reply Sources/App/configure.swift]: * variables

File: Sources/App/Store/CRUDStore.swift (hunk 1 above)
Comment (portellaa) [reply Sources/App/Store/CRUDStore.swift:19]: i moved all them to the same line. i think it's easy to read, you start a reactive stream from that point, so i think it's easy to understand. no?

File: Sources/App/configure.swift (hunk 2 above)
Comment (portellaa) [reply Sources/App/configure.swift:22]: this is just an example, so it's ok to leave the credentials here. those are not even correct.

regarding the inject them as env vars, yes, i'm doing that in marmoto

**Pull request merged**